### PR TITLE
Remove unused property (#2290)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `HCA` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or commented that sandbox will be skipped</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [ ] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

**Review by achave11 — APPROVED** [review]
Looks good, I would have done it in 2 commits. One, the removal of the property. Two, enhancing the documentation. I'll approve and leave it at your discretion.

File: src/azul/tdr.py [hunk 1]
```diff
@@ -142,6 +142,10 @@ def get_access_token(self) -> str:
 class TDRClient(SAMClient):
     """
     A client for the Broad Institute's Terra Data Repository code-named Jade.
+
+    Azul currently has no need to access the TDR API, but we expect it to
+    become necessary in the future. For now, this class is only used to verify
```
Comment (achave11) [inline src/azul/tdr.py:147]: Remember to enforce [code style](https://github.com/DataBiosphere/azul/blob/develop/CONTRIBUTING.rst#1code-style) second bullet point.

File: src/azul/tdr.py [hunk 2]
```diff
@@ -142,6 +142,10 @@ def get_access_token(self) -> str:
 class TDRClient(SAMClient):
     """
     A client for the Broad Institute's Terra Data Repository code-named Jade.
+
+    Azul currently has no need to access the TDR API, but we expect it to become
+    necessary in the future. For now, this class is only used to verify that we
```
Comment (hannes-ucsc) [inline src/azul/tdr.py:147]: The future has arrived. The PR that adds it is really close to landing: #2267. I think we can remove this paragraph.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: src/azul/tdr.py (hunk 2 above)
Comment (noah-aviel-dove) [reply src/azul/tdr.py]: #2267 has been merged but `TDRClient` is still not used for any purpose other than access verification.

Comment (hannes-ucsc) [reply src/azul/tdr.py]: The Azul service lambda, access TDR's DRS implementation and it needs to authenticate for that. This paragraph is incorrect and needs to be removed.

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**